### PR TITLE
modules: mcux: introduce CONFIG_MCUX_CORE_SUFFIX

### PR DESCRIPTION
**Pull request by danieldegrasse**
MCUX SDK depends on "core suffixes" in order to identify which core a build is targeting on a multicore part. Previously, this information was parsed from the CONFIG_SOC string, but with hardware model v2 this is no longer possible. Introduce the Kconfig MCUX_CORE_SUFFIX, which multicore SOCs can set to inform MCUX which core the build is targeting.

### CODE REVIEW

File: modules/Kconfig.mcux [hunk 1]
```diff
@@ -10,6 +10,12 @@ config HAS_MCUX
 
 if HAS_MCUX
 
+config MCUX_CORE_SUFFIX
+	string
+	help
+	  String describing the core identifer used by MCUX SDK when using
+	  dual core parts
```
Comment (nordicjm) [inline modules/Kconfig.mcux:17]: I posted this in the other PR but why is this needed, why can't your module check for `CPU_CORTEX_A53` (or similar)?

**Comment by danieldegrasse** [comment]
> so the idea is that MCUX_CORE_SUFFIX is a string representation of cpuclusters, like PART_NUMBER or SOC for PART_NUMBER_XX and SOC_XX respectively? shouldn't be this generic and not specific to MCUX

Yes, that's the idea- I made this specific to MCUX because I felt that the strings our SDK chose to describe cpuclusters were sometimes a bit arbitrary- IE the LPC parts use `_cm33_core0` and `_cm33_core1` for each core, or the LPC54114 uses `_cm4` and `_cm0plus`. If you'd like it to be generic, I have no problem adding a generic Kconfig instead.

File: modules/Kconfig.mcux (hunk 1 above)
Comment (danieldegrasse) [reply modules/Kconfig.mcux:17]: The module could check for this- but it results in an (IMO) ugly `if` and `elseif` ladder in the module, because we don't always use the same suffix even if the core is the same.

IE on our dual core cm33 parts, we use suffixes like `_cm33_core0` and `_cm33_core1`, but on dual core parts that have an Xtensa core and CM33, the suffix would be `_cm33`. The end result of this is an `if`/`elseif` ladder like this:

```
if (CONFIG_CPU_LPC55S69_CPU0)

elseif(CONFIG_CPU_LPC55S69_CPU1)

elseif(CONFIG_CPU_MIMXRT595S_CM33)

....
```

Comment (tejlmand) [reply modules/Kconfig.mcux:17]: Naturally when combining two systems, like Zephyr's Kconfig and a HAL like NXP's, (which is probably designed externally to Zephyr) then some way of mapping between Zephyr and the HAL is needed.

And on the positive side it's good that this happens in the module's glue code.
But I take this new Kconfig and the old (mis-?)use of the CONFIG_SOC name as a sign that the existing implementation / mapping has not been done in a very clean way. (Ref: https://github.com/zephyrproject-rtos/hal_nxp/blob/d45b14c198d778658b7853b48378d2e132a6c4be/mcux/CMakeLists.txt#L15-L50)

Usually I wouldn't be fond of dealing with a Kconfig `SUFFIX` setting like this, but I understand the need for a mapping, and keeping this isolated to the glue layer makes it acceptable.

However, on the long term I think it would be nice to investigate if this could be made even cleaner.

**Comment by tejlmand** [comment]
> If you'd like it to be generic, I have no problem adding a generic Kconfig instead.

Generic solutions are always welcome 🙂
So a solution where we make the CPU cluster name available in Kconfig, similar to what we do with SoC name sounds reasonable,

Just remember that in such case we should also ensure CPU cluster name in Kconfig is aligned with its soc.yml counterpart, just like we try to do with SoC names.

**Comment by JiafeiPan** [comment]
@danieldegrasse, the patch in this PR has been merged by through https://github.com/zephyrproject-rtos/zephyr/pull/69400/commits
